addons/macrochanger/macrochanger.lua

```diff
@@ -26,143 +27,67 @@
-		g:write('Disable All: 0\n')
-		g:write('WAR Book: 1\nWAR Page: 1\nMNK Book: 2\nMNK Page: 1\nWHM Book: 3\nWHM Page: 1\nBLM Book: 4\nBLM Page: 1\nRDM Book: 5\nRDM Page: 1\nTHF Book: 6\nTHF Page: 1\n')
-		g:write('PLD Book: 7\nPLD Page: 1\nDRK Book: 8\nDRK Page: 1\nBST Book: 9\nBST Page: 1\nBRD Book: 10\nBRD Page: 1\nRNG Book: 11\nRNG Page: 1\nSAM Book: 12\nSAM Page: 1\n')
-		g:write('NIN Book: 13\nNIN Page: 1\nDRG Book: 14\nDRG Page: 1\nSMN Book: 15\nSMN Page: 1\nBLU Book: 16\nBLU Page: 1\nCOR Book: 17\nCOR Page: 1\nPUP Book: 18\nPUP Page: 1\n')
-		g:write('DNC Book: 19\nDNC Page: 1\nSCH Book: 20\nSCH Page: 1\nGEO Book: 20\nGEO Page: 1\nRUN Book: 20\nRUN Page: 1\n')
-		g:close()
-		DisableAll = 0
-        macros = {
-            WAR = {Book = '1', Page = '1'},
-            MNK = {Book = '2', Page = '1'},
-            WHM = {Book = '3', Page = '1'},
-            BLM = {Book = '4', Page = '1'},
-            RDM = {Book = '5', Page = '1'},
-            THF = {Book = '6', Page = '1'},
-            PLD = {Book = '7', Page = '1'},
-            DRK = {Book = '8', Page = '1'},
-            BST = {Book = '9', Page = '1'},
-            BRD = {Book = '10', Page = '1'},
-            RNG = {Book = '11', Page = '1'},
-            SAM = {Book = '12', Page = '1'},
-            NIN = {Book = '13', Page = '1'},
-            DRG = {Book = '14', Page = '1'},
-            SMN = {Book = '15', Page = '1'},
-            BLU = {Book = '16', Page = '1'},
-            COR = {Book = '17', Page = '1'},
-            PUP = {Book = '18', Page = '1'},
-            DNC = {Book = '19', Page = '1'},
-            SCH = {Book = '20', Page = '1'},
-            GEO = {Book = '20', Page = '1'},
-            RUN = {Book = '20', Page = '1'},
-            }
-		print('Default settings file created')
-		windower.add_to_chat(12,'MacroChanger created a settings file and loaded!')
-	else
-		f:close()
-		for curline in io.lines(windower.addon_path..'data/settings.txt') do
-			local splat = curline:gsub(':',''):split(' ')
-			local cmd = ''
-			if splat[1] and macros[splat[1]:upper()] and splat[2] ~=nil and (splat[2]:lower() == 'book' or splat[2]:lower() == 'page') and splat[3] then
-				macros[splat[1]:upper()][splat[2]:ucfirst()] = splat[3] -- Instead of a number, this can also be 'disabled'
-			elseif splat[1] and splat[2] and (splat[1]..' '..splat[2]) == 'disable all' and tonumber(splat[3]) then
-				globaldisable = tonumber(splat[3])
-			end
-		end
-		windower.add_to_chat(12,'MacroChanger read from a settings file and loaded!')
-	end
+function addon_command (...)
+    local args = {...}
+    local i = 1
+    while i < #args + 1 do
+        if args[i] == 'help' then
+            options.help()
+			return
+        elseif options[args[i]] then
+            options[args[i]](args[i+1])
+            i = i + 2
+        else
+            windower.add_to_chat(17, 'Unknown parameter: '..args[i])
+            options.help()
+            return
+        end
+    end
```

Comment:
Is this mechanism meant to loop over multiple options in the same line? While that wouldn't really be an issue, it prevents passing an arbitrary number of arguments to the respective functions (although it seems that this addon in particular doesn't need that). And you don't really gain much from it, since you can easily chain commands with `;`.